src/azmi-main/Operations.cs

```diff
@@ -205,5 +210,46 @@ public string setBlob_byBlob(string filePath, string blobUri, string identity =
                 throw IdentityError(identity, ex);
             }
         }
+
+        // Validate URL to a key vault secret is generally valid
+        private void validateSecretURL(string secretIdentifierUrl, out Uri keyVaultUri, out Uri secretIdentifierUri)
+        {
+            // Example of expected URL: https://mine-key-vault.vault.azure.net/secrets/mineSecret.pwd
+            if (!Uri.IsWellFormedUriString(secretIdentifierUrl, UriKind.Absolute))
+                throw new UriFormatException($"Provided URL '{secretIdentifierUrl}' is not well formed URL.");
+
+            secretIdentifierUri = new Uri(secretIdentifierUrl);
+
+            if (secretIdentifierUri.Scheme != Uri.UriSchemeHttps)
+                throw new UriFormatException($"Only '{Uri.UriSchemeHttps}' protocol is supported.");
+
+            // e.g. http://mine-key-vault.vault.azure.net
+            keyVaultUri = new Uri(secretIdentifierUri.GetLeftPart(UriPartial.Authority));
+            // Segments = /, secrets/, mineSecret.pwd
+            if (secretIdentifierUri.Segments.Count() <= 2)
+                throw new UriFormatException($"URL '{secretIdentifierUrl}' is missing a path to secret.");
+        }
+
+        // Get secret from key vault
+        public string getSecret(string secretIdentifierUrl, string identity = null)
+        {
+            Uri keyVaultUri, secretIdentifierUri;
+            this.validateSecretURL(secretIdentifierUrl, out keyVaultUri, out secretIdentifierUri);
+
+            string secretName = secretIdentifierUri.Segments.Last();
```

Comment:
❗️  Why don't you include `secretName` as output from validation method? Like this, I have feeling variable `secretIdentifierUri` is introduced without reason in main method.

Also, I think you can declare out variables directly in call to method, like `out var keyVaultUri`